rasa/server.py

```diff
@@ -599,6 +606,58 @@ def create_app(
 
         return response.json(response_body)
 
+    @app.post("/conversations/<conversation_id>/inject-intent")
+    @requires_auth(app, auth_token)
+    @ensure_loaded_agent(app)
+    async def inject_intent(request: Request, conversation_id: Text):
+        request_params = request.json
+
+        intent_to_inject = request_params.get("name", None)
+        entities = request_params.get("entities", [])
+
+        if not intent_to_inject:
+            raise ErrorResponse(
+                400,
+                "BadRequest",
+                "Name of the intent not provided in request body.",
+                {"parameter": "name", "in": "body"},
+            )
+
+        verbosity = event_verbosity_parameter(request, EventVerbosity.AFTER_RESTART)
+
+        try:
+            async with app.agent.lock_store.lock(conversation_id):
+                tracker = get_tracker(app.agent, conversation_id)
+                output_channel = _get_output_channel(request, tracker)
+                if intent_to_inject not in app.agent.domain.intents:
+                    raise ErrorResponse(
```

Comment:
This exception raised here will be catched by the surrounding `try...except`, is this intentional? The error code "ValueError" will be lost.

Reply:
did you validate that @JEM-Mosig ? That should be caught by the unit tests, shouldn't it?